src/components/GlobalHeader/RightContent.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import { FormattedMessage, formatMessage } from 'umi/locale'
+import { Spin, Tag, Menu, Icon, Avatar, Tooltip } from 'antd'
+import moment from 'moment'
+import groupBy from 'lodash/groupBy'
+import { NoticeIcon } from 'ant-design-pro'
+import HeaderSearch from '../HeaderSearch'
+import HeaderDropdown from '../HeaderDropdown'
+import SelectLang from '../SelectLang'
+import styles from './index.less'
+
+export default function GlobalHeaderRight (props) {
+  const {
+    currentUser,
+    fetchingNotices,
+    onNoticeVisibleChange,
+    onMenuClick,
+    onNoticeClear,
+    theme,
+    notices = [],
+    dispatch
+  } = props
+
+  function getNoticeData () {
+    if (notices.length === 0) {
+      return {}
+    }
+    const newNotices = notices.map(notice => {
+      const newNotice = { ...notice }
```

Comment:
hmm, I don't understand this line, how the spread operator (`...notice`) works here?

Reply:
I debugged and checked the value of states here. They use the same value and reference. 

- [ ] Changed `newNotice` to `notice`

- [ ]  commented this condition statement.

```
if (newNotice.datetime) {
   newNotice.datetime = moment(notice.datetime).fromNow()
}
```